### PR TITLE
preload_models.py script downloads the weight files

### PR DESCRIPTION
**Pull request by lstein**
## The `preload_models.py` script now interactively downloads popular SD weights and VAE files.

This uses the code and logic of PR #1189 

- user can select which weight files to download
- download management uses the huggingface_hub api
- user must log in to huggingface, generate an access token, and accept license terms the very first time this is run. After that, everything works automatically.
- added placeholder for docs for installing models
- also got rid of unused config files. hopefully they weren't needed for textual inversion, but I don't think so.

Typical session:
```
$ ./scripts/preload_models.py 
Welcome to InvokeAI. This script will help download the Stable Diffusion weight files
and other large models that are needed for text to image generation. At any point you may interrupt
this program and resume later.

Would you like to download the Stable Diffusion model weights now? [y] 

Choose the weight file(s) you wish to download. Before downloading you 
will be given the option to view and change your selections.

[1] stable-diffusion-1.5:
    The newest Stable Diffusion version 1.5 weight file (4.27 GB) (recommended)
    Download? [y] 
[2] inpainting-1.5:
    RunwayML SD 1.5 model optimized for inpainting (4.27 GB) (recommended)
    Download? [y] 
[3] stable-diffusion-1.4:
    The original Stable Diffusion version 1.4 weight file (4.27 GB) 
    Download? [n] 
[4] waifu-diffusion-1.3:
    Stable Diffusion 1.4 fine tuned on anime-styled images (4.27) 
    Download? [n] y
[5] ft-mse-improved-autoencoder-840000:
    StabilityAI improved autoencoder fine-tuned for human faces (recommended; 335 MB) (recommended)
    Download? [y] 
The following weight files will be downloaded:
   [1] stable-diffusion-1.5*
   [2] inpainting-1.5
   [4] waifu-diffusion-1.3
   [5] ft-mse-improved-autoencoder-840000
*default
Ok to download? [y] 

To download the Stable Diffusion weight files you need to read and accept the
CreativeML Responsible AI license. If you have not already done so, please 
create an account at https://huggingface.co. Then login under your account and
read and accept the license available at https://huggingface.co/CompVis/stable-diffusion-v-1-4-original.

Press <enter> when you are ready to continue:

Thank you! Now you need to authenticate with your HuggingFace access token.
Go to https://huggingface.co/settings/tokens and create a token. Copy it to the
clipboard and paste it here: 
Password: 
 * v1-5-pruned-emaonly.ckpt: exists
 * sd-v1-5-inpainting.ckpt: exists
 * model-epoch09-float32.ckpt: downloading or retrieving from cache...
Downloading: 100%|█████████████████████████████████████████████████████████████| 4.27G/4.27G [00:41<00:00, 102MB/s]
 * vae-ft-mse-840000-ema-pruned.ckpt: exists
Successfully installed stable-diffusion-1.5, inpainting-1.5, waifu-diffusion-1.3, ft-mse-improved-autoencoder-840000
Successfully created new configuration file ./configs/models.yaml
Installing bert tokenizer (ignore deprecation errors)......success
Installing Kornia requirements......success
Loading CLIP model......success
Installing models from RealESRGAN and facexlib......success
Loading models from GFPGAN
Installing CodeFormer model file......success
Installing clipseg model for text-based masking......success
Installing safety model for NSFW content detection......success

```

### CODE REVIEW

**Comment by mauwii** [comment]
is there a switch to run noninteractive, since this would break all automations, or at least `.github/workflows/test-invoke-conda.yml`, `docker-build/build.sh`, the Dockerfile itself (and maybe others I don't know atm)?

File: docs/installation/INSTALLING_MODELS.md [hunk 1]
```diff
@@ -0,0 +1,261 @@
...
+
+The predefined configuration file for InvokeAI (located at
+`configs/models.yaml`) provides entries for each of these weights
+files. `stable-diffusion-1.5` is the default model used, and we
+strongly recommend that you install this weights file if nothing else.
+
+## Community-Contributed Models
+
+There are too many to list here and more are being contributed every
+day. [This page](https://rentry.org/sdmodels) hosts an updated list of
```
Comment (tildebyte) [inline docs/installation/INSTALLING_MODELS.md:50]: That rentry list is really sketchy - I don't necessarily have a problem with that. I *do* think that we should also include https://huggingface.co/sd-concepts-library as a source of more... reputable... fine-tuned models

Comment (lstein) [reply docs/installation/INSTALLING_MODELS.md:50]: Good feedback. I've added a `--no-interactive` switch that restores the previous behavior. It doesn't interact with user, and it also doesn't download weight files. I think github actions are doing weight file downloading separately.

Comment (lstein) [reply docs/installation/INSTALLING_MODELS.md:50]: > That rentry list is really sketchy - I don't necessarily have a problem with that. I _do_ think that we should also include https://huggingface.co/sd-concepts-library as a source of more... reputable... fine-tuned models

Are we able to load .bin files? I haven't written any code to do that.

Comment (Any-Winter-4079) [reply docs/installation/INSTALLING_MODELS.md:50]: Yes, we can (at least could a while ago) load .bin (for embeddings at least).

Comment (lstein) [reply docs/installation/INSTALLING_MODELS.md:50]: I'm learning something new every day! I checked out the --embedding_path option for the first time, and sure enough I can load one of those concept .bin files.

It looks to me as though we ought to be able to load multiple .bin files. Does this not work?

Comment (lstein) [reply docs/installation/INSTALLING_MODELS.md]: > > That rentry list is really sketchy - I don't necessarily have a problem with that. I _do_ think that we should also include https://huggingface.co/sd-concepts-library as a source of more... reputable... fine-tuned models

I've added the link to the Hugging Face concepts library. Oddly, this page doesn't load on Linux Chrome. I had to go to Firefox to view it.

Comment (Any-Winter-4079) [reply docs/installation/INSTALLING_MODELS.md]: I don't know how to load both at the same time. But it'd be useful.
I tried (which understandably only uses the last one)
`python scripts/invoke.py --embedding_path learned_embeds-2.bin --embedding_path uglysonic.bin`
And this
`python scripts/invoke.py --embedding_path learned_embeds-2.bin uglysonic.bin`
But both fail.

Comment (lstein) [reply docs/installation/INSTALLING_MODELS.md]: Internally the code seems to handle multiple embedding paths, but perhaps one will replace the previous one. I'll give it a try.

**Comment by lstein** [comment]
Because I need to have this for the 1-click installer. I'm going to merge into dev now and make the release candidate.

**Comment by n00mkrad** [comment]
Why no fp16 download options? 99% of people are running fp16 anyway

**Pull request merged**